chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
+				let id = toId(target[j]);
+				if (MafiaData.roles[id]) {
+					if (typeof MafiaData.roles[id] === 'string') {
+						target[j] = MafiaData.roles[MafiaData.roles[id]].name;
+						id = toId(target[j]);
+					}
+					if (role.id) {
+						hasProblem = true;
+						problems.push(`The role "${role.name}" has multiple possible base roles (${MafiaData.roles[id].id} or ${role.baseRole})`);
+						break;
+					}
+					role = Object.assign({}, MafiaData.roles[id]);
+					role.baseRole = toId(role.name);
+					role.name = target.slice().map(p => { return toId(p) === 'solo' ? '' : p; }).join(' ');
+					role.id = toId(role.name);
+					if (!role.memo) role.memo = [];
+					role.memo.unshift('During the Day, you may vote for whomever you want lynched.');
+					target.splice(j, 1);
+					j++;
+				}
+			}
+			if (hasProblem) continue;
+			if (!role.name) role = {name: target.join(' '), id: toId(target.join('')), memo: []};
+			// Add modifiers
+			for (let j = 0; j < target.length; j++) {
+				let id = toId(target[j]);
+				if (id.endsWith('shot') && parseInt(target[j].substring(0, 1))) {
+					// Special case for X-Shot modifiers
+					let number = target[j].substring(0, 1);
+					let memo = MafiaData.modifiers.xshot.memo.slice();
+					memo = memo.map(m => { return m.replace(/X/g, number); });
+					role.memo = role.memo.concat(memo);
+					target.splice(j, 1);
+					j--;
+				} else if (MafiaData.modifiers[id]) {
+					if (typeof MafiaData.modifiers[id] === 'string') target[j] = MafiaData.modifiers[id];
+					if (MafiaData.modifiers[id].memo) role.memo = role.memo.concat(MafiaData.modifiers[id].memo);
+					target.splice(j, 1);
+					j--;
+				}
+			}
+			if (hasProblem) continue;
+			// Determine the role's alignment
+			for (let j = 0; j < target.length; j++) {
+				let id = toId(target[j]);
+				if (MafiaData.alignments[id]) {
+					if (typeof MafiaData.alignments[id] === 'string') target[j] = MafiaData.alignments[id];
+					if (role.alignment) {
+						// A role cant have multiple alignments
+						hasProblem = true;
+						problems.push(`The role "${role.name}" has multiple possible alignments (${MafiaData.alignments[id].id} or ${role.alignment})`);
+						break;
+					}
+					role.alignment = MafiaData.alignments[id].id;
+					role.memo = role.memo.concat(MafiaData.alignments[id].memo);
+					if (!role.image && MafiaData.alignments[id].image) role.image = MafiaData.alignments[id].image;
+					target.splice(j, 1);
+					j--;
+				}
+			}
+			if (!role.alignment) role.alignment = 'town'; // Default to town
+			if (alignments.indexOf(role.alignment) === -1) alignments.push(role.alignment);
+			// Handle anything that is unknown
+			if (target.length) {
+				role.memo.push(`To learn more about your role, PM the host (${this.host}).`);
+			}
+			this.originalRoles.push(role);
+		}
+		if (alignments.length < 2) problems.push(`There must be at least 2 different alignments in a game!`);
+		if (problems.length) {
```

Comment:
Should probably have a way to force setting this.

Reply:
I'm not a fan of force setting the roles, as it can create issues such as the script not really knowing which alignment to give to a role (ex: Mafia Alien or something like that), which can cause accidental leakage of information (your partners are: mafia1 mafia2 and notmafia (oops)).

Reply:
What about things like UCTs with totally nonstandard roles or bastard mods though?

Reply:
Also, themes like SS2 could be affected (as there is possibility of only 1 alignment in the game)

Reply:
True, though it would probably be best if the server considers all players are in the "solo" alignment, and dose not give role info. This information would be left for the host to provide via PMs like normal with UTCs.

Reply:
Did this ^